### PR TITLE
Move tests into respective modules

### LINKED ISSUE #194: Move tests into respective modules
**Is your feature request related to a problem? Please describe.**
#193 extracted the rule implementations into separate modules in preparation for #181. The tests for different components are still in `/tests`, which precludes individual testing of modules.

**Describe the solution you'd like**
Move files in [`/tests`](https://github.com/tweag/rules_nixpkgs/tree/b39b20edc4637032bc65f6a93af888463027767c/tests) into respective module directories, such that each module can be tested separately with `bazel test //...`.

**Describe alternatives you've considered**
Keep everything at the top level and only test together. This has the downside that modules cannot be tested before publishing to BCR at some point in the future.

### PR DESCRIPTION
**Pull request by fricklerhandwerk**
depends on #212
closes #194

### CODE REVIEW

File: WORKSPACE [hunk 1]
```diff
@@ -1,61 +1,34 @@
...
-http_archive(
-    name = "io_bazel_stardoc",
-    sha256 = "6d07d18c15abb0f6d393adbd6075cd661a2219faab56a9517741f0fc755f6f3c",
-    strip_prefix = "stardoc-0.4.0",
-    urls = [
-        "https://mirror.bazel.build/github.com/bazelbuild/stardoc/archive/0.4.0.tar.gz",
-        "https://github.com/bazelbuild/stardoc/archive/0.4.0.tar.gz",
-    ],
-)
+load("@io_tweag_rules_nixpkgs//nixpkgs:repositories.bzl", "rules_nixpkgs_dependencies")
```
Comment (avdv) [inline WORKSPACE:7]: you load the `rules_nixpkgs_dependencies` twice, first on line 3 and here again.

**Review by avdv — CHANGES_REQUESTED** [review]
Overall, LGTM. Just found a nit, and please resolve the conflict in `.github/workflows/workflow.yaml`

BTW, I also tested these changes with the daml project again, just to be safe.

**Comment by fricklerhandwerk** [comment]
@avdv Thanks for the hints, fixed.

**Review by avdv — APPROVED** [review]

**Review by aherrmann — APPROVED** [review]
LGTM, Thank you!

**Pull request merged**